assets/js/components/wp-dashboard/WPDashboardPopularPages.js

```diff
@@ -39,45 +44,69 @@ import { numFmt } from '../../util';
 const { useSelect } = Data;
 
 const WPDashboardPopularPages = ( { WidgetReportZero, WidgetReportError } ) => {
-	const dateRangeDates = useSelect( ( select ) =>
-		select( CORE_USER ).getDateRangeDates( {
+	const { report, titles, loading, error } = useSelect( ( select ) => {
+		const dateRangeDates = select( CORE_USER ).getDateRangeDates( {
 			compare: true,
 			offsetDays: DATE_RANGE_OFFSET,
-		} )
-	);
+		} );
+		const reportArgs = {
+			...dateRangeDates,
+			metrics: [
+				{
+					expression: 'ga:pageviews',
+					alias: 'Pageviews',
+				},
+			],
+			dimensions: [ 'ga:pagePath' ],
+			orderby: [
+				{
+					fieldName: 'ga:pageviews',
+					sortOrder: 'DESCENDING',
+				},
+			],
+			limit: 5,
+		};
+		const data = {
+			report: select( MODULES_ANALYTICS ).getReport( reportArgs ),
+			error: select( MODULES_ANALYTICS ).getErrorForSelector(
+				'getReport',
+				[ reportArgs ]
+			),
+			loading: true,
+		};
+		const reportLoaded = select(
+			MODULES_ANALYTICS
+		).hasFinishedResolution( 'getReport', [ reportArgs ] );
 
-	const reportArgs = {
-		...dateRangeDates,
-		metrics: [
-			{
-				expression: 'ga:pageviews',
-				alias: 'Pageviews',
-			},
-		],
-		dimensions: [ 'ga:pageTitle', 'ga:pagePath' ],
-		orderby: [
-			{
-				fieldName: 'ga:pageviews',
-				sortOrder: 'DESCENDING',
-			},
-		],
-		limit: 5,
-	};
+		let pagePaths = [];
+		let hasLoadedPageTitles = true;
+		if ( reportLoaded ) {
+			const { startDate, endDate } = dateRangeDates;
+			const pageTitlesArgs = {
+				startDate,
+				endDate,
+			};
+			( data.report?.[ 0 ]?.data?.rows || [] ).forEach(
+				( { dimensions } ) => {
+					pagePaths = pagePaths.concat(
+						dimensions.filter(
+							( url ) => ! pagePaths.includes( url )
+						)
+					);
+				}
+			);
```

Comment:
This is not safe to rely on the `dimensions` property as on an array with URLs only. In our case, only the first item in the dimensions array is interesting for us. Please, rework this loop to use the `dimensions[0]` element only. The same applies to other components as well, please, update them all.

Reply:
https://github.com/google/site-kit-wp/pull/3958/commits/a52d5da689077a9cbc62b42c918026c5bc15e3bd